src/shared/state/compactBlock.ts

```diff
@@ -0,0 +1,97 @@
+'use client';
```

Comment:
issue (conventions): use kebab-case in the file naming

Reply:
suggestion: move the files from `state` to `shared/api`. Feels like the right choice for methods requesting data